SweetTweetApp/src/main/java/posts/PostsFunctions.java

```diff
@@ -0,0 +1,126 @@
+package posts;
+
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+public class PostsFunctions {
```

Comment:
`PostsFunctions` is not a good name. It implies the class is just a compilation of utility methods, merely "functions".
But it has a *state*, Merely the collection of  "posts"
So technically this is rather a DAO - data access object.

Still there's another significant problem.

Your "posts" are a static field (why?). But all the accessors of "posts" are not static (also, why?) This is not right. Static fields (save for constants) must be accessed via static methods, per-instance fields must be accessed via per-istance methods, and not in any other way. Down below in other classes you create an instance of "PostFunctions" Also, what for? Even if you create a thousand of  `new PostsFunctions()... new PostsFunctions()...` - they will all share the same "posts" field, and this confusing and buggy